lib/runtime/allocator/freeing_bump.go

```diff
@@ -505,14 +505,17 @@ func (f *FreeingBumpHeapAllocator) Deallocate(mem runtime.Memory, ptr uint32) (e
 func bump(bumper *uint32, size uint32, mem runtime.Memory) (uint32, error) {
 	requiredSize := uint64(*bumper) + uint64(size)
 
-	if requiredSize > uint64(mem.Size()) {
+	if requiredSize > mem.Size() {
 		requiredPages, ok := pagesFromSize(requiredSize)
 		if !ok {
-			return 0, fmt.Errorf("%w: required size %d dont fit uint32",
-				ErrAllocatorOutOfSpace, requiredSize)
+			panic(fmt.Sprintf("cannot calculate number of pages from size %d", requiredSize))
```

Comment:
why are we panicking here and not just returning an error?

Reply:
fixed the `requiredPages` calculation to return an error if required size does not fit in uint32, but in the `currentPages` calculation we should panic since we should be able to have the correct page size and if not that is a critical problem.

ref: https://github.com/paritytech/polkadot-sdk/blob/46ba85500ffc77fa8e267c5f38b2c213550d68fa/substrate/client/allocator/src/freeing_bump.rs#L519

Also changed the `pagesIncrease` check to log an error if pages don't increase

Reply:
oh they use a debug assert.  To the best of my knowledge rust debug asserts are not actually checked on production builds, but I don't think we can replicate that.

Reply:
yup, then the best option is log